assert/assertions_test.go

```diff
@@ -2807,12 +2807,31 @@ func TestNeverTrue(t *testing.T) {
 	False(t, Never(mockT, condition, 100*time.Millisecond, 20*time.Millisecond))
 }
 
-func TestEventuallyIssue805(t *testing.T) {
+// Check that a long running condition doesn't block Eventually.
+// See issue 805 (and its long tail of following issues)
+func TestEventuallyTimeout(t *testing.T) {
 	mockT := new(testing.T)
 
 	NotPanics(t, func() {
-		condition := func() bool { <-time.After(time.Millisecond); return true }
+		t.Log("start")
+		done, done2 := make(chan bool), make(chan bool)
```

Comment:
It's idiomatic to use a `struct{}` instead of `bool` as it takes no space.
```suggestion
		done, done2 := make(chan struct{}), make(chan struct{})
```

Reply:
Thanks for the catch.

Fixed (amending my commit).